nixos/lib/testing/nodes.nix

```
@@ -159,10 +160,17 @@ in
 
     passthru.nodes = config.nodesCompat;
 
-    defaults = mkIf config.node.pkgsReadOnly {
-      nixpkgs.pkgs = config.node.pkgs;
-      imports = [ ../../modules/misc/nixpkgs/read-only.nix ];
-    };
+    defaults = mkMerge [
+      (mkIf (config.node.pkgs != null) {
+        nixpkgs.pkgs = config.node.pkgs;
+      })
+      (mkIf (config.node.pkgs == null) {
+        nixpkgs.pkgs = hostPkgs;
```

```suggestion
        nixpkgs.pkgs = mkDefault hostPkgs;
```

Probably should be this, so that modules can override it without getting a priority conflict.

---

Done.